src/main/java/org/benetech/servicenet/repository/ActivityRepository.java

```diff
@@ -96,15 +96,43 @@ public ActivityRecord findOneByOrganizationId(UUID orgId) {
 
         queryCriteria.select(selectRoot).distinct(true);
 
-        addFilters(queryCriteria, selectRoot, ownerId, searchName, searchOn, filtersActivityDTO);
+        addFilters(queryCriteria, selectRoot, ownerId, searchName, searchOn, filtersActivityDTO, false);
         addSorting(queryCriteria, pageable.getSort(), selectRoot);
 
         CriteriaQuery<Long> countCriteria = cb.createQuery(Long.class);
         Root<ActivityInfo> selectRootCount = countCriteria.from(ActivityInfo.class);
 
         countCriteria.select(cb.countDistinct(selectRootCount));
 
-        addFilters(countCriteria, selectRootCount, ownerId, searchName, searchOn, filtersActivityDTO);
+        addFilters(countCriteria, selectRootCount, ownerId, searchName, searchOn, filtersActivityDTO, false);
+
+        List<ActivityInfo> results = em.createQuery(queryCriteria)
+            .setFirstResult((int) pageable.getOffset())
+            .setMaxResults(pageable.getPageSize())
+            .getResultList();
+
+        Long total = em.createQuery(countCriteria).getSingleResult();
+
+        return new PageImpl<>(results, pageable, total.intValue());
+    }
+
+    public Page<ActivityInfo> findPartnerRecordsWithFilters(UUID ownerId, String searchName, SearchOn searchOn,
```

Comment:
what is the difference between those two methods? is it only the parameter passed to the addFilters?
is the separate method necessary in this case?

Reply:
done